commands/apps.go

```diff
@@ -293,49 +305,74 @@ func (c *AppsInitCmd) InitializeNodeBasicApp(stdout, stderr bytes.Buffer) (err e
 	checkPkgJSON, err := os.Open("package.json")
 	if err != nil {
 		log.Println("[WARN] package.json does not exist. Creating package.json")
-		cmd := exec.Command("npm", "init", "-y")
-		cmd.Stdout = &stdout
-		cmd.Stderr = &stderr
-		err := cmd.Run()
+		err := c.CreatePkgJson(stdout, stderr)
 		if err != nil {
 			log.Println("[ERROR] There was an error creating package.json. Is node installed?")
-			panic(err)
+			return fmt.Errorf("there was an error creating package.json. Is node installed? %w", err)
 		} else {
 			log.Println("[INFO] package.json created")
 		}
 	}
 	defer checkPkgJSON.Close()
 	validPkgJSON, err := os.OpenFile("package.json", os.O_RDWR, 0777)
 	if err != nil {
-		panic(err)
+		return fmt.Errorf("failed to open package.json %w", err)
 	}
+	defer validPkgJSON.Close()
 	log.Println("[INFO] Validating package.json")
-	finfo, err := validPkgJSON.Stat()
+	buf, err := ioutil.ReadFile("package.json")
 	if err != nil {
-		panic(err)
+		return fmt.Errorf("failed to read package.json %w", err)
 	}
-	buf := make([]byte, finfo.Size())
-	_, err = validPkgJSON.Read(buf)
+	fStr := string(buf)
+	if len(fStr) == 0 {
+		err := os.Remove("package.json")
+		if err != nil {
+			log.Println("[ERROR] unable to remove empty package.json")
+		}
+		log.Println("[WARN] package.json is empty. Creating package.json")
+		err = c.CreatePkgJson(stdout, stderr)
+		if err != nil {
+			log.Println("[ERROR] There was an error creating package.json. Is node installed?")
```

Comment:
We should not log and return an error - it's either one or the other. Wrap it and return it. Then some caller in the chain will log it.